cdc/entry/mounter.go

```diff
@@ -316,22 +316,28 @@ func ParseDDLJob(tblInfo *model.TableInfo, rawKV *model.RawKVEntry, id int64) (*
 
 // parseJob unmarshal the job from "v".
 func parseJob(v []byte, startTs, CRTs uint64) (*timodel.Job, error) {
-	job := &timodel.Job{}
-	err := json.Unmarshal(v, job)
+	var job timodel.Job
+	err := json.Unmarshal(v, &job)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	log.Debug("get new DDL job", zap.String("detail", job.String()))
+	// TiDB write DDL Binlog for every DDL Job,
+	// we must ignore jobs that are cancelled or rollback
+	// For older version TiDB, it writes DDL Binlog in the txn
+	// that the state of job is changed to *synced*
+	// Now, it writes DDL Binlog in the txn that the state of
+	// job is changed to *done* (before change to *synced*)
+	// At state *done*, it will be always and only changed to *synced*.
```

Comment:
These comments here is a bit confusing for me to read.

Reply:
it's from other code snippet.